atom/browser/ui/cocoa/atom_access_controller.mm

```diff
@@ -0,0 +1,194 @@
+#include "atom/browser/ui/cocoa/atom_access_controller.h"
+
+#import <AVFoundation/AVFoundation.h>
+#import <Cocoa/Cocoa.h>
+
+@implementation AtomAccessController
+
++ (instancetype)sharedController {
+  static dispatch_once_t once;
+  static AtomAccessController* sharedController;
+  dispatch_once(&once, ^{
+    sharedController = [[self alloc] init];
+  });
+  return sharedController;
+}
+
+- (instancetype)init {
+  if ((self = [super init])) {
+    if (@available(macOS 10.14, *)) {
+      switch (
+          [AVCaptureDevice authorizationStatusForMediaType:AVMediaTypeVideo]) {
+        case AVAuthorizationStatusAuthorized:
+        case AVAuthorizationStatusRestricted:
+          cameraAccessState_ = AccessStateGranted;
+          break;
+        case AVAuthorizationStatusDenied:
+          cameraAccessState_ = AccessStateDenied;
+          break;
+        case AVAuthorizationStatusNotDetermined:
+          cameraAccessState_ = AccessStateUnknown;
+      }
+      switch (
+          [AVCaptureDevice authorizationStatusForMediaType:AVMediaTypeAudio]) {
+        case AVAuthorizationStatusAuthorized:
+        case AVAuthorizationStatusRestricted:
+          microphoneAccessState_ = AccessStateGranted;
+          break;
+        case AVAuthorizationStatusDenied:
+          microphoneAccessState_ = AccessStateDenied;
+          break;
+        case AVAuthorizationStatusNotDetermined:
+          microphoneAccessState_ = AccessStateUnknown;
+      }
+      [[[NSWorkspace sharedWorkspace] notificationCenter]
+          addObserver:self
+             selector:@selector(applicationLaunched:)
+                 name:NSWorkspaceDidLaunchApplicationNotification
+               object:nil];
+    } else {
+      // access is always allowed pre-10.14 Mojave
+      cameraAccessState_ = AccessStateGranted;
+      microphoneAccessState_ = AccessStateGranted;
+    }
+  }
+  return self;
+}
+
+// pops up an alert to change mic prefs (restart required to take effect)
+- (void)alertForMicrophoneAccess {
+  if (microphoneAccessState_ == AccessStateDenied) {
+    NSAlert* alert = [[NSAlert alloc] init];
+    alert.alertStyle = NSAlertStyleWarning;
+    alert.messageText = @"This app needs access to the microphone.";
```

Comment:
This should probably be customizable somehow, as with the strings beneath it

Reply:
i was planning to open this as a follow-up PR i think 🤔